File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // If ./caesar is not followed by 1 other string, ask the user to do so and quit program
    if (argc != 2 )
    {
        printf("Enter 1 integer after ./caesar\n");
        return 1;
    }
    int key = atoi(argv[1]);
    string plaintext = get_string("plaintext: ");
    printf("ciphertext: ");
    for (int i = 0, n = strlen(plaintext); i < n ; i++)
    {
        // Checks to see if each character is uppercase letter
        if isupper((plaintext[i]))
        {
            // Converts from ascii index to alphabetical index by subtracting, then converts back to ascii
            plaintext[i] -= 65;
            plaintext[i] = (plaintext[i] + (key)) % 26 + 65;
            printf("%c", plaintext[i]);
        }
        // Checks to see if each character is lowercase letter
        else if (islower(plaintext[i]))
        {
            // Converts from ascii index to alphabetical index by subtracting, then converts back to ascii
            plaintext[i] -= 97;
            plaintext[i] = (plaintext[i] + (key)) % 26 + 97;
            printf("%c", plaintext[i]);
        }
        // If not uppercase nor lowercase letter, must be character that should not be encrypted, so simply print
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");
    return 0;
}

File: caesar.py
import cs50
import sys


def main():
    alphalen = 26
    # If python caesar.py is not followed by 1 other string, ask the user to do so and quit program
    if len(sys.argv) != 2:
        print("Enter 1 integer after caesar.py")
        exit(1)
    key = int(sys.argv[1])
    # Acquire plaintext from user
    plaintext = cs50.get_string("Plaintext: ")
    print("ciphertext: ", end="")
    length = len(plaintext)
    for i in plaintext:
        # If character is alphabetic, convert to alphabetical index, add key, convert back, and print
        if i.isalpha():
            if i.isupper():
                letter = ord(i) - ord('A')
                i = (letter + (key)) % alphalen + ord('A')
            elif i.islower():
                letter = ord(i) - ord('a')
                i = (letter + (key)) % alphalen + ord('a')
            print(chr(i), end="")
        # If character is not alphabetic, then simply print
        else:
            print(i, end="")
    print()


if __name__ == "__main__":
    main()
